### ISSUE TITLE
Problems running angel_predict.py

### ISSUE THREAD

**Issue by steffi778** [PacificBiosciences/ANGEL#26]
Hello Magdoll,

I try to run ANGEL. All the steps until angel_predict.py working fine but then I get errors.
I have IsoSeq data from different rice cultivars and used as training data set the cDNA file of the reference Nipponbare (IRGSPv1.36).

When I run angel_predict.py (either for all or for the single split files) I always get the same error in the end:

Traceback (most recent call last):
  File "/home/mpimp-golm.mpg.de/schaarschmidt/.conda/envs/anaCogent/bin/angel_predict.py", line 4, in <module>
    __import__('pkg_resources').run_script('Angel==2.4', 'angel_predict.py')
  File "/home/mpimp-golm.mpg.de/schaarschmidt/.local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 738, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/home/mpimp-golm.mpg.de/schaarschmidt/.local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1499, in run_script
    exec(code, namespace, namespace)
  File "/home/mpimp-golm.mpg.de/schaarschmidt/.conda/envs/anaCogent/lib/python2.7/site-packages/Angel-2.4-py2.7-linux-x86_64.egg/EGG-INFO/scripts/angel_predict.py", line 22, in <module>
    distribute_ANGEL_predict(args.fasta_filename, args.output_prefix, args.classifier_pickle, args.cpus, args.min_angel_aa_length, args.min_dumb_aa_length, args.use_rev_strand, args.output_mode, args.max_angel_secondORF_distance)
  File "/home/mpimp-golm.mpg.de/schaarschmidt/.conda/envs/anaCogent/lib/python2.7/site-packages/Angel-2.4-py2.7-linux-x86_64.egg/Angel/SmartORF.py", line 288, in distribute_ANGEL_predict
    pool.map(ANGEL_predict_worker_helper, data)
  File "/home/mpimp-golm.mpg.de/schaarschmidt/.conda/envs/anaCogent/lib/python2.7/multiprocessing/pool.py", line 253, in map
    return self.map_async(func, iterable, chunksize).get()
  File "/home/mpimp-golm.mpg.de/schaarschmidt/.conda/envs/anaCogent/lib/python2.7/multiprocessing/pool.py", line 572, in get
    raise self._value
KeyError: 'X'

Every step is performed as you describe in your tutorial so far. I also tried to run it just at one core, but it  failed. Additionally, I got the same error in max depth (as reported in  #24)  when I am running the test data which you have provided.

I hope you can help and thank you very much already
steffi

**Comment by Magdoll** [comment]
Hi @steffi778 

Will you be willing to share a single split file input (which will include the fasta and the trained pickle) with me over email confidentially?

If so please leave an email so I can request a file upload.

Thanks,
--Liz

**Comment by steffi778** [comment]
Hi Liz!
Sure I can do this. Could you send the link to Schaarschmidt@mpimp-golm.mpg.de?

Thanks,
steffi

**Comment by Magdoll** [comment]
Hi @steffi778  Request sent.

**Comment by Magdoll** [comment]
Hi @steffi778 ,

I believe I fixed the problem. THe issue came from sequences in the input that contains non A/T/C/G/ bases, in this case 'N'. By default ANGEL now converts them to a 'A' base. After this fix I am able to finish running your sample input.

The new ANGEL is now v2.5.

--Liz

**Comment by CapricornMM** [comment]
hello@Magdoll ,

When I use angel_predict.py, I found that angel_predict.py hangs and could nothing suddenly, then I run each of the split up fasta files. Finally, I found that several sequences in each split file can't be predicted. These sequences can't get its cds, pep, utr files. This is the reason for angel_predict.py hangs. I want to know why these sequences can't be predicted. And how can I solve it? 

Looking forward to your reply.

--Meimei
